Show Mautic admin page titles

Yame- opened the pull request:
| Q | A |
| --- | --- |
| Enhancement | Yes |
| Issues addressed (#s or URLs) | #1142 |
#### Description:

Changes the title tag so that it's more informative instead of only "Mautic".
It will now read "Mautic - Email - [name_of_email]" or "Mautic - Campaigns"
#### Steps to test this PR:
1. Install this PR
2. Click on some pages and see of the title changes correctly

Yame- commented:
Hi @jordan8037310, 
I've read that you were going to deliver a fix as well. 
Am I going in the right direction with this implementation?
Any advice would be awesome! 👍

jordan8037310 commented:
@Yame- this looks good! I actually started on the other side of this problem and hadn't worked on the Javascript side yet. 

I would argue that a better order is `headerTitle - Mautic` instead of vice versa. Reasoning being if you have a lot of tabs open they all shorten and then you can only see `Mautic - `

Yame- commented:
Makes more sense @jordan8037310!

jordan8037310 commented on app/bundles/CoreBundle/Assets/js/1.core.js:
```diff
@@ -1173,6 +1174,23 @@ var Mautic = {
+        if( -1 !== route.indexOf('view') ){
+            //loading view of module title
+            var currentModule = route.split('/')[3];
+            mQuery('title').html( 'Mautic - ' + currentModule[0].toUpperCase() + currentModule.slice(1) + ' - ' + mQuery('.page-header h3').html() );
```
1186: rearrange to `pageTitle | Mautic`

jordan8037310 commented on app/bundles/CoreBundle/Assets/js/1.core.js:
```diff
@@ -1173,6 +1174,23 @@ var Mautic = {
+            mQuery('title').html( 'Mautic - ' + currentModule[0].toUpperCase() + currentModule.slice(1) + ' - ' + mQuery('.page-header h3').html() );
+        } else {
+            //loading basic title
+            mQuery('title').html( 'Mautic - ' + mQuery('.page-header h3').html() );
```
Rearrange to Page Title before Mautic

jordan8037310 commented on app/bundles/CoreBundle/Views/Default/head.html.php:
```diff
@@ -10,7 +10,11 @@
-    <title><?php echo $view['slots']->get('pageTitle', 'Mautic'); ?></title>
+    <title><?php echo $view['slots']->get('pageTitle', 'Mautic'); ?>
+	<?php if( !empty($view['slots']->get('headerTitle', '')) ): ?>
+		- <?php echo $view['slots']->get('headerTitle', ''); ?>
```
Rearrange to Page Title before `Mautic`. Also a pipe may be a better choice, smaller character so more real estate on tab for other characters, e.g. `Calendar | Mautic`

Yame- commented:
Thanks for your suggestions @jordan8037310.
I've implemented them in my last commit.

alanhartless commented:
I like this! Only issue is on some pages, the page header contains HTML (like on a contact detail's page).

![](http://alan.direct/drop/2016-10-11_01-47-30.png)

Can we fix that?

Yame- commented:
Thanks for the catch @alanhartless.

I've added some more code to catch those html tags and remove them.
Also, I've added a title update when an item gets saved.

escopecz commented:
It looks and works good! :+1: 

Just one question, why there is "S" on every detail page title? Like `S | Contact Name Here | Mautic` or `S | Form Name | Mautic`?

Yame- commented:
Hi @escopecz, 
Can't replicate your "S" problem. Detail pages seem fine on my end. 
I'm running latest version of Mautic. I've checked Chrome, Firefox and Safari. 

Any idea what could be the reason of this? 🤔
